resources/templates/cloudformation/auto_scaling_group.yaml

```diff
@@ -0,0 +1,87 @@
+AWSTemplateFormatVersion: 2010-09-09
```

Comment:
This is the Cloud Formation template. The Go templating is used to set the type of ASG e.g. master or worker. The block device mappings are also templated.

Otherwise the values are provided using Cloud Formation parameters.